feat(atomic-react): add iife output for atomic react + publish to CDN (#2426)
coveo/ui-kit: packages/atomic-react/README.md

`npm i @coveo/atomic-react`

> ### Alternate installation -- CDN

Reviewer: ```suggestion
### Alternate installation — CDN
```
(you can't tell with the GitHub editor, but the dash I'm using it the length of two small dashes. It's an em dash, to be precise)

> The library is also available from Coveo's CDN as an `IIFE` (Immediately Invoked Function Expression). This is an approach that we recommend only for quick prototyping and testing, and that can be used with very minimal front end tooling and build tools.

Reviewer: ```suggestion
The library is also available in the Coveo CDN as an `IIFE` (Immediately Invoked Function Expression). This is an approach that we recommend only for quick prototyping and testing, and that can be used with very minimal front-end tooling and build tools.
```

> The `IIFE` approach in production is discouraged - the browser has to download the entire library code, regardless of which components are actually used. Using a bundler (for example Webpack) is recommended.

Reviewer: ```suggestion
We recommend against using the `IIFE` approach in production. The browser has to download the entire library code, regardless of which components are actually used. We rather recommend using a bundler in this situation (for example Webpack).
```

> You can read more on this approach down below.

Reviewer: ```suggestion
You can read more about this approach [below](#usage-with-cdn-scripts--iife-).
```
(not totally sure about the link though, and it depends on you accepting my suggestion below 😉)

  }}
></AtomicSearchInterface>;
```

> ## Usage with CDN scripts

Reviewer: ```suggestion
## Usage with CDN Scripts (IIFE)
```
(I think)

> For this approach to work, different scripts need to be pulled in the page in the correct order, using proper external dependencies with matching versions.

Reviewer: ```suggestion
For this approach to work, you need to pull different scripts in the page in the correct order, using proper external dependencies with matching versions.
```

First, identify the version of `@coveo/atomic` that is used by `@coveo-atomic-react`.

> This can be done with `npm view @coveo/atomic-react`, and identify the `@coveo/atomic` dependencies:

Reviewer: ```suggestion
You can do that by running `npm view @coveo/atomic-react`, and then identifying the `@coveo/atomic` dependency:
```

```
npm view @coveo/atomic-react

@coveo/atomic-react@1.23.11 | Proprietary | deps: 1 | versions: 209
React specific wrapper for the Atomic component library
https://github.com/coveo/ui-kit#readme

dist
.tarball: https://registry.npmjs.org/@coveo/atomic-react/-/atomic-react-1.23.11.tgz
.shasum: 64ba0a5f686e3638b1180910f6fed5e4980bd9e2
.integrity: sha512-/qI5O7SWNinBYWAedRW62ko0oF4c/hfapgqKe9xSq/NJfg+KBzcR3SbdyhpaUAWpPX1910GP9cMyr4wf8irJnQ==
.unpackedSize: 771.1 kB

dependencies:
@coveo/atomic: 1.108.2
```

> In the above example, we can see that of the time of this writing, the current latest version of `@coveo/atomic-react` is `1.23.11`, and that it is using `@coveo/atomic` at version `1.108.2`.

Reviewer: ```suggestion
In the above example, you can see that of the time of this writing, the current latest version of `@coveo/atomic-react` is `1.23.11`, and that it is using `@coveo/atomic` at version `1.108.2`.
```

> Then, the same need to be done for the matching version of `@coveo/headless` used by the matching `@coveo/atomic` version.

Reviewer: ```suggestion
Then, you need to do the same to find the matching version of `@coveo/headless` used by the target`@coveo/atomic` version.
```

Using the above example with `@coveo/atomic@1.108.2`:

```bash
npm view @coveo/atomic@1.108.2

@coveo/atomic@1.108.2 | Apache-2.0 | deps: 15 | versions: 728
A web-component library for building modern UIs interfacing with the Coveo platform
https://docs.coveo.com/en/atomic/latest/

dist
.tarball: https://registry.npmjs.org/@coveo/atomic/-/atomic-1.108.2.tgz
.shasum: 6e844073f55f10328b4b828c0190b8fe31403fbf
.integrity: sha512-LhFR6k7NdGi8pYHmWOq3gM3rE0zeTKc6biTa1Z+ZKr5TEXIr97rpxi1HSnViQUowQYYCcY44dvsAYz2yyPvXTA==
.unpackedSize: 87.1 MB

dependencies:
@coveo/bueno: 0.42.1                  @stencil/store: 1.5.0                 focus-visible: 5.2.0
@coveo/headless: 1.103.3              coveo-styleguide: 9.34.4              i18next-http-backend: 1.4.1
@popperjs/core: ^2.11.6               dayjs: 1.11.5                         i18next: 20.6.1
@salesforce-ux/design-system: ^2.16.1 dompurify: 2.3.10                     stencil-inline-svg: 1.1.0
@stencil/core: 2.17.3                 escape-html: 1.0.3                    ts-debounce: ^4.0.0

```

> In the above example, we can see that that `@coveo/atomic@1.108.2` has a dependency on `@coveo/headless@1.103.3`.

Reviewer: ```suggestion
In the above example, you can see that `@coveo/atomic@1.108.2` has the dependency `@coveo/headless@1.103.3`.
```

> In summary, as of the time of this writing, the versions that we should keep in mind are the following:

Reviewer: ```suggestion
To summarize, as of the time of this writing, the versions that we should keep in mind are the following:
```

```html
<head>
  <!-- React and ReactDOM need to be included  -->
  <script crossorigin src="https://unpkg.com/react@18/umd/react.production.min.js"></script>
  <script crossorigin src="https://unpkg.com/react-dom@18/umd/react-dom.production.min.js"></script>

  <!-- Optional script, which allows to use JSX directly in an inline script in the page -->
  <script crossorigin src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>

  <!-- @coveo/headless need to be included as a dependency -->
  <!-- Note the matching major and minor version as explained above -->
  <script crossorigin src="https://static.cloud.coveo.com/headless/v1.103/headless.js"></script>

  
  <!-- @coveo/atomic need to be included as a dependency -->
  <!-- Note the matching major and minor version as explained above -->
  <script crossorigin type="module" src="https://static.cloud.coveo.com/atomic/v1.108/atomic.esm.js"></script>
  
  <!--And then finally @coveo/atomic-react is included -->
  <!-- Note the matching major and minor version as explained above -->
  <script crossorigin src="https://static.cloud.coveo.com/atomic-react/v1.23/iife/atomic-react.min.js"></script>
</head>
```

> Once this is done, you can now start using `CoveoAtomicReact` directly with an inline script tag: 

Reviewer: ```suggestion
Once this is done, you can start using `CoveoAtomicReact` directly with an inline script tag: 
```
(since you already set a time frame with `Once this is done`)
